stock_warehouse_flow/views/stock_warehouse_flow.xml

```diff
@@ -0,0 +1,189 @@
+                                widget="domain"
+                                options="{'model': 'stock.move', 'in_dialog': true}"
+                            />
+            </group>
+            <group name="to" string="To">
+              <field
+                                name="delivery_steps"
+                                attrs="{'invisible': [('delivery_route_id', '!=', False)]}"
+                            />
+              <field
+                                name="sequence_prefix"
+                                attrs="{'invisible': [('delivery_route_id', '!=', False)], 'required': [('delivery_route_id', '=', False)]}"
+                            />
+              <field
+                                name="delivery_route_id"
+                                attrs="{'invisible': [('delivery_route_id', '=', False)]}"
+                            />
+              <field
+                                name="to_picking_type_id"
+                                attrs="{'invisible': [('delivery_route_id', '=', False)]}"
+                            />
+              <field
+                                name="warning"
+                                class="alert alert-warning"
+                                role="alert"
+                                attrs="{'invisible': [('warning', '=', False)]}"
+                                nolabel="1"
+                                colspan="2"
+                            />
+              <button
+                                name="action_generate_route"
+                                type="object"
+                                string="Generate route"
+                                class="oe_highlight"
+                                attrs="{'invisible': [('delivery_route_id', '!=', False)]}"
+                            />
+            </group>
+          </group>
+          <notebook>
+            <page string="Rules" name="rules">
+              <field name="rule_ids" nolabel="1">
+                <tree>
+                  <field name="action" />
+                  <field name="picking_type_id" />
+                  <field name="location_src_id" />
+                  <field name="location_id" />
+                  <field name="group_id" />
+                </tree>
+              </field>
+            </page>
+            <page string="Impacted routes" name="impacted_routes">
+              <field name="impacted_route_ids" nolabel="1">
+                <tree>
+                  <field name="name" />
+                </tree>
+              </field>
+            </page>
+            <page
+                            string="Technical Information"
+                            name="technical_info"
+                            groups="base.group_no_one"
+                        >
+              <group>
+                <group string="Locations">
+                  <field name="pack_stock_loc_id" readonly="1" />
```

Comment:
Can you make locations & picking types readonly only when route is generated (route_id != False) ?

Reply:
And also based on the number of steps:
1 step, I can define out
2 steps, I can define pick & out + out location
3 steps, I can define pick, pack & out + pack & out locations